### PR TITLE
Fix package dependencies in setup

### PR DESCRIPTION
**Pull request by roaffix**
# Description

- Add missing `install_requires` option in setup config. Add only required packages (without dev packages like `coverage`, `pylint` etc.).
- Add some extra `classifiers` to the setup to cover more search options on [PyPI](pypi.org).
- Minor setup and `.gitignore` "code style" changes:
   - remove template comments from `setup.py`;
   - move all config from `setup.py` to `setup.cfg`;
   - add comments for ignored directories and files in `.gitignore`.

Fixes # (issue)

#116 

## Type of change

- [x] Bug fix (non-breaking change which fixes an issue)

### CODE REVIEW

File: setup.py [hunk 1]
```diff
@@ -1,207 +1,3 @@
...
-    # what's used to render the link text on PyPI.
-    project_urls={  # Optional
-        'Documentation': 'https://mlfinlab.readthedocs.io/en/latest/',
-        'Bug Reports': 'https://github.com/hudson-and-thames/mlfinlab/issues',
-        'Project Boards': 'https://github.com/orgs/hudson-and-thames/projects',
-        'Source': 'https://github.com/hudson-and-thames/mlfinlab',
-        'Blog': 'https://hudsonthames.org/blog/',
-    },
-
-    # Create package
```
Comment (Jackal08) [inline setup.py]: We need to please keep these comments:
```
    # Create package
    # python setup.py bdist_wheel
    # python3 -m twine upload --repository-url https://test.pypi.org/legacy/ dist/*  (This is the test repo)
    # twine upload dist/*  (This is official repo)
```
It helps me to push the python package to PyPi index.

File: setup.cfg [hunk 2]
```diff
@@ -0,0 +1,62 @@
...
+    education
+
+[options]
+include_package_data = True
+packages = find:
+python_requires = 
+    ~=3.6
+setup_requires = 
+    setuptools
+install_requires =
```
Comment (Jackal08) [inline setup.cfg]: Is there a way to point these requirements to the requirements.txt file? That way we don't need to manually update the values.

File: setup.py [hunk 3]
```diff
@@ -1,207 +1,3 @@
-# Great link to learn to build a package:
-# https://packaging.python.org/guides/distributing-packages-using-setuptools/#choosing-a-versioning-scheme
```
Comment (Jackal08) [inline setup.py]: Can we please keep this comment:
```
# https://packaging.python.org/guides/distributing-packages-using-setuptools/#choosing-a-versioning-scheme
```

**Comment by Jackal08** [comment]
Please pull dev branch into this branch to update this branch with base branch.

File: setup.cfg (hunk 2 above)
Comment (roaffix) [reply setup.cfg]: There are two different requirements. In `install_requires` you should put minimal requirements, only necessary ones. It's well described [here](https://packaging.python.org/discussions/install-requires-vs-requirements/).

File: setup.py (hunk 1 above)
Comment (roaffix) [reply setup.py]: It's unnecessary comments, so I'm not sure that this is a great idea to keep it within a package. I think it'd be better to keep it somewhere locally.

**Comment by cmackenziek** [comment]
Getting this fixed would be great for the library. Currently pretty cumbersome to install all dependencies manually. If you need help I'd be glad to help but it seems this PR has it covered?

**Comment by Jackal08** [comment]
> Getting this fixed would be great for the library. Currently pretty cumbersome to install all dependencies manually. If you need help I'd be glad to help but it seems this PR has it covered?

Hi @cmackenziek, 

It would be great if you would like to issue a PR for this. I would like to keep the inline comments in the setup.py file. I realise they add to the overhead but I constantly reference them and would like to keep them. 

A big reason this ticket is delayed is because of the additional overhead for me to figure out what the best practices are for setting this up. roaffix did do a very good job, I just haven't revisited this.

A PR would be welcomed. (An emphasis of best practices.)

**Comment by cmackenziek** [comment]
OK @Jackal08 I'll take a look!

**Comment by rspadim** [comment]
I really liked, agree

**Comment by Jackal08** [comment]
This PR was included here: https://github.com/hudson-and-thames/mlfinlab/pull/187

Thank you to @roaffix and @cmackenziek for the work you did here.

**Pull request closed without merging**